### PR TITLE
DAC6-3263

### PR DESCRIPTION
**Pull request by koladeadewuyi**
This PR implements the backend APIs for listing and viewing financial institutions associated with a subscriptionId.
To enable testing via cURL the authentication is not implemented yet and will be done when integrated with the frontend as part of DAC6-3255

### CODE REVIEW

File: test-common/uk/gov/hmrc/crsfatcafimanagement/SpecBase.scala [hunk 1]
```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
```
Comment (oarshad) [inline test-common/uk/gov/hmrc/crsfatcafimanagement/SpecBase.scala:2]: 2024

File: app/uk/gov/hmrc/crsfatcafimanagement/models/common/ResponseDetails.scala [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.crsfatcafimanagement.models.common
+
+import play.api.libs.json.{Json, OFormat}
+import uk.gov.hmrc.crsfatcafimanagement.models.FIDetail
+
+final case class ResponseDetails(FIDetails: List[FIDetail])
```
Comment (oarshad) [inline app/uk/gov/hmrc/crsfatcafimanagement/models/common/ResponseDetails.scala:22]: same as above

File: app/uk/gov/hmrc/crsfatcafimanagement/models/common/ResponseCommon.scala [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.crsfatcafimanagement.models.common
+
+import play.api.libs.json.{Json, OFormat}
+import uk.gov.hmrc.crsfatcafimanagement.models.RequestType
+
+final case class ResponseCommon(
+  OriginatingSystem: String,
```
[the remark this answers is not in the document]
Comment (matthewharris1) [reply app/uk/gov/hmrc/crsfatcafimanagement/models/common/ResponseCommon.scala:23]: I'd suggest as its a dummy spec and to keep with consistency with other services we have this isn't necessary YET? Once we know for sure what CADX specs is going to look like we can implement something if casing doesn't pair up with scala standards? 

That would be my suggestion anyways

[the remark this answers is not in the document]
Comment (oarshad) [reply app/uk/gov/hmrc/crsfatcafimanagement/models/common/ResponseCommon.scala:23]: in that case we can keep all these as camelCase for now?

[the remark this answers is not in the document]
Comment (matthewharris1) [reply app/uk/gov/hmrc/crsfatcafimanagement/models/common/ResponseCommon.scala:23]: Rather match whats on the dummyspec than not, it doesn't cause any errors or compilation issues?

**Review by oarshad — DISMISSED** [review]

File: app/uk/gov/hmrc/crsfatcafimanagement/models/common/ResponseCommon.scala (hunk 3 above)
[the remark this answers is not in the document]
Comment (koladeadewuyi) [reply app/uk/gov/hmrc/crsfatcafimanagement/models/common/ResponseCommon.scala:23]: The value names in the specs are indeed unusual. Ideally the `PascalCase` pattern returned by `CADX` ought be transformed by `EIS` before sending the response to us. Perhaps @jagankumar-hmrc  could persuade them to do this transformation.
Using the JsonNaming as suggested by @oarshad would be a good middle ground.

**Review by C-Thomp — APPROVED** [review]
👍

**Pull request merged**